### PR TITLE
FIX AMARI SHITCODE ALLOWING HUMANS TO BE THRALLED AFTER DEATH

### PR DESCRIPTION
**Pull request by k-k-karma**

### CODE REVIEW

File: hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm [hunk 1]
```diff
@@ -292,7 +292,33 @@
 				to_chat(target, "<span class='userdanger'>You wrest yourself away from [user]'s hands and compose yourself</span>")
 				enthralling = 0
 				return
-
+		if(!in_range(user, target))
```
Comment (ghost) [inline hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm:295]: do_after will make sure they're in range

File: hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm [hunk 2]
```diff
@@ -292,7 +292,33 @@
...
+			to_chat(user, "<span class='warning'>You need to be closer to enthrall [target]!</span>")
+			revert_cast()
+			return
+		if(!target.key || !target.mind)
+			to_chat(user, "<span class='warning'>The target has no mind!</span>")
+			revert_cast()
+			return
+		if(target.stat)
```
Comment (ghost) [inline hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm:303]: I can just use the `Sleep` verb while being enthralled in order to escape?

File: hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm [hunk 3]
```diff
@@ -292,7 +292,33 @@
...
+			return
+		if(is_shadow_or_thrall(target))
+			to_chat(user, "<span class='warning'>You can not enthrall allies!</span>")
+			revert_cast()
+			return
+		if(!ishuman(target))
+			to_chat(user, "<span class='warning'>You can only enthrall humans!</span>")
+			revert_cast()
+			return
+		if(enthralling)
```
Comment (ghost) [inline hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm:315]: Are you stupid?

File: hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm [hunk 4]
```diff
@@ -292,7 +292,33 @@
...
+			to_chat(user, "<span class='warning'>You are already enthralling!</span>")
+			revert_cast()
+			return
+		if(!target.client)
```
Comment (ghost) [inline hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm:319]: I can just disconnect while being enthralled in order to escape?

Comment (k-k-karma) [reply hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm]: you can do that beforehand

File: hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm (hunk 3 above)
Comment (k-k-karma) [reply hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm]: i copy and pasted

File: hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm (hunk 2 above)
Comment (k-k-karma) [reply hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm]: they can wake you up

File: hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm (hunk 4 above)
Comment (ghost) [reply hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm:319]: Shouldn't be able to escape once it starts.

File: hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm (hunk 2 above)
Comment (ghost) [reply hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm:303]: not if you sleep at the literal last second

**Comment by Kawaii-Big-Boss** [comment]
Why not just remove it from rotation instead?

**Pull request closed without merging**